text/0000-dyn-error-generic-member-access.md

```diff
@@ -259,84 +264,66 @@ impl<'r, 'out> Request<'r, 'out> {
     ///
     /// This method can be chained within `provide` implementations using the
     /// `?` operator to concisely provide multiple objects.
-    pub fn provide_with<T: ?Sized + 'static, F>(mut self, cb: F) -> ProvideResult<'r, 'out>
+    pub fn provide_with<T: ?Sized + 'static, F>(&mut self, cb: F) -> &mut Self
```

Comment:
```suggestion
    pub fn provide_with<T: ?Sized + 'static, F>(self: Pin<&mut Self>, cb: F) -> Pin<&mut Self>
```

Reply:
this line no longer compiles when you apply this change.

```rust
            let this = unsafe { &mut *(self as *mut _ as *mut RequestBuf<Option<&'out T>>) };
```

do you wanna try applying these changes in https://github.com/yaahc/nostd-error-poc/ first and open a PR there, once it's merged there I'll copy the changes into the RFC.